generator/internal/request_id.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/internal/request_id.h"
+#include <google/api/field_info.pb.h>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+
+bool MeetsRequestIdRequirements(
+    google::protobuf::FieldDescriptor const& descriptor) {
+  if (descriptor.type() != google::protobuf::FieldDescriptor::TYPE_STRING ||
+      descriptor.is_repeated() || descriptor.has_presence()) {
+    return false;
+  }
+  if (!descriptor.options().HasExtension(google::api::field_info)) return false;
+  auto const info = descriptor.options().GetExtension(google::api::field_info);
+  return info.format() == google::api::FieldInfo::UUID4;
+}
+
+std::string RequestIdFieldName(
+    YAML::Node const& service_config,
+    google::protobuf::MethodDescriptor const& descriptor) try {
```

Comment:
I don't think so. I have included an item for discussion in today's meeting, in case somebody has a reason to support builds without exceptions I have not thought of.

Basically the generator needs to compile on a limited number of builds (*San builds, clang-tidy builds, and the builds that use it).  Running without exception support is not a requirement.

Reply:
Sounds good to me, but then we should probably remove the conditionals from `generator/internal/printer.h`.

Reply:
Done in #13611